universe/node/api_users.go

```diff
@@ -43,7 +43,24 @@ func (n *Node) apiUsersGetMe(c *gin.Context) {
 		return
 	}
 
-	userDTO := api.ToUserDTO(userEntry, true)
+	userTypeAttributeValue, ok := n.GetNodeAttributeValue(
+		entry.NewAttributeID(universe.GetSystemPluginID(), universe.Attributes.Node.GuestUserType.Name),
+	)
+	if !ok || userTypeAttributeValue == nil {
+		err := errors.New("Node: apiUsersGetMe: failed to get user type attribute value")
+		api.AbortRequest(c, http.StatusInternalServerError, "server_error", err, n.log)
+		return
+	}
+
+	guestUserType := utils.GetFromAnyMap(*userTypeAttributeValue, universe.Attributes.Node.GuestUserType.Key, "")
+	guestUserTypeID, err := uuid.Parse(guestUserType)
+	if err != nil {
+		err := errors.New("Node: apiUsersGetMe: failed to parse guest user type id")
+		api.AbortRequest(c, http.StatusInternalServerError, "server_error", err, n.log)
+		return
+	}
```

Comment:
looks like we are copypasting the same code several times
I think it's better to move this logic to some helper function

Reply:
Duplicated code moved to method:
```go
func (n *Node) GetGuestUserTypeID() (uuid.UUID, error) {
	userTypeAttributeValue, ok := n.GetNodeAttributeValue(
		entry.NewAttributeID(universe.GetSystemPluginID(), universe.Attributes.Node.GuestUserType.Name),
	)
	if !ok || userTypeAttributeValue == nil {
		err := errors.New("failed to get user type attribute value")
		return uuid.Nil, err
	}

	guestUserType := utils.GetFromAnyMap(*userTypeAttributeValue, universe.Attributes.Node.GuestUserType.Key, "")
	guestUserTypeID, err := uuid.Parse(guestUserType)
	if err != nil {
		err := errors.New("failed to parse guest user type id")
		return uuid.Nil, err
	}

	return guestUserTypeID, err
}
```

Maybe it can be generalised to something like this
```
func GetNodeAttributeValue[T any](a universe.Attribute, defaultValue T)(T, error) {

}
```
But this is for future discussion

Reply:
for me it's a bit better to move it to our helpers and call universe.GetNode() under the hood
but you may leave it as it is